### DISCUSSION TITLE
Digital Gas Control w/ Phidget OUT1000

### DISCUSSION THREAD

**Discussion by djbetterly** [artisan-roaster-scope/artisan#713] [Q&A] [answered]
I've had the OUT1000 setup for quite some time, and I've always struggled with good gas control within Artisan. Without being critical, as I'm no programmer, I'm curious why the control within Artisan can't be more precise?  If I use the Phidget Control Panel and manually adjust the OUT1000 voltage, I can get extremely precise gas adjustments. Given that, shouldn't it be possible within Artisan to have more precision?  Maybe I'm doing something wrong as well...I'm open to feedback.

I'm just looking for more information, and curious if this is possible.

Also, I'm using this in conjunction with the DAQ1400.

I appreciate the help and guidance.

**Comment by MAKOMO** [comment]
Could give a little more information what you mean by "more precise"? Artisan, as the Phidget Control Panel, just advices the device via its API and the Phidget driver to change its output voltage. There is nothing that can be done different then to say "set output to 3V". Both, Artisan and the Phidget Control Panel are using the same API and driver.

Did you use a voltage meter to check if the output of the OUT1000 is reaching the intended voltage levels that you send via Artisan (or the Control Panel)?

I do not have the OUT1000 (12bit, 0-4.2V), but I am using a OUT1002 (16bit, -10-10V, isolated) module to control my gas output and that just works as intended.

Not sure why you mention the DAQ1400 as this seems to me irrelevant which other modules you have connected.

Maybe you can give a little more context on your hardware setup, like the gas valve or valve control board you are connecting to the OUT1000 and what this assumes as input.

**Comment by djbetterly** [comment, marked as answer]
Thank you for getting back to me so promptly.  I suspect some of the lack of precision is in the valve itself, I know proportion valves do have certain limitations. 

My hardware setup is a proportion valve from IQvalves.com.  The valve is a 12v valve with a .125" orifice ([spec sheet](https://4cde57fd-4e59-4042-8a4b-75626fcdaa2f.filesusr.com/ugd/a4cf14_d2b94eb8f6f44ab893b28cd5293a4b52.pdf)), and the controller is their proprietary controller which runs on a 0-5v signal ([spec sheet](https://4cde57fd-4e59-4042-8a4b-75626fcdaa2f.filesusr.com/ugd/a4cf14_93bc754a95f841fc838d2e70f5cc8f83.pdf)).  The valves are connected to Artisan via the OUT1000.

Regarding voltage measurements. I've measured the voltage using the Phidget Control Panel, the control panel output and the OUT1000 match accordingly.

Now in Artisan, and I suspect maybe this is where the problem lies, I have the slider output set as follows:
Event: Burner
Action:  VOUT Command
Command: out(0,{})
Offset: 0
Factor: 0.050
Min: 30
Max: 80
Bernoulli, Coarse, Temp, Unit all have nothing checked or input.

When the slider is set to 80 (max) I get a full 4v, at 30 I get 1.5v (the minimum to keep the flame on).

Now in terms of precision, I was wrong, Artisan is actually quite precise.  When I use the Phidget Control Panel I can be a bit more precise as I can adjust the voltage in smaller increments (0.001), where it seems Artisan only adjusts in 0.5v increments - unless I'm doing something wrong.  This becomes a problem on the lower end of gas pressure, high end is fine.  

In the end I guess it comes down the mechanics of the valve as well. 

Which valve do you use?

Please let me know if I've left anything unanswered.

Again, I appreciate your time!

**Reply by MAKOMO** [reply]
The only limitation here is your valve and the sliders range. The slider event values are always integers, by default between 0 and 100 (usually interpreted in percent. Not sure why you are not using the full range, but limit it to 30 to 80. Of course this limits your increments to 0.05V (not 0.5V as you write). Setting the sliders min to 0 and max to 100, using offset 1.5 and factor 0.025 results in increments of 0.025. You can as well increase the sliders range to min 0 to max 200, offset again to 1.5 and factor to 0.0125 which gives you increments of 0.125. So you can get all the precision you need from the Artisan software. Of course our OUT1000 Phidget is limited to 12bit resolution. Not sure thus why you are not using its full voltage range up to 5V.

I cannot comment on the quality of your valve, but there are no principle limitations on proportion valves I am aware of. The valve I am using operates very accurately for this purpose, but is not available for general purchase.

**Reply by MAKOMO** [reply]
>  I've measured the voltage using the Phidget Control Panel

Thus you did not measure the output voltage of the module, but just looked at what the Phidget Control Panel reported that it did send as command to the module. Not exactly the same. Actually, even close to the same. Totally different.

**Comment by djbetterly** [comment]
Thank you.  So I do have the OUT1002 as well, but I originally was having better luck with the OUT1000, I’m guessing that was mostly due to lack of knowledge on how to properly set it up in Artisan.

If I were I to put that back into play, what settings would you recommend for artisan?

Also, can you help me understand how those settings are calculated?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by MAKOMO** [reply]
For using the OUT1002 with your 0-5V valve, you would first put it in 5V mode, eg. with a command action on the ON button (see under 4.1 on https://artisan-scope.org/devices/phidgets/ for details).

<img width="1063" alt="Screenshot 2021-10-26 at 13 48 39" src="https://user-images.githubusercontent.com/13909011/138873104-ad960520-e570-4e92-b8a0-032f8914f9ed.png">

and then choose a factor of 0.05 for translating the 0-100 slider values into 0-5V commands towards the device.

<img width="1063" alt="Screenshot 2021-10-26 at 13 49 54" src="https://user-images.githubusercontent.com/13909011/138873090-ca11dacb-1a35-4a19-a085-53777a10d9e2.png">

It might also work to send the voltage range with each slider command as follows, but I did not test this and I do not know what happens to the output voltage the moment the voltage range is changed.

<img width="1063" alt="Screenshot 2021-10-26 at 13 58 18" src="https://user-images.githubusercontent.com/13909011/138873073-c660c159-8f2e-4a48-a41a-f4af1bc59d39.png">

Excel/OpenOffice provide the SLOPE and INTERCEPT functions that take a list of target values (0 and 5 in this case) and a list of source values (0 and 100 here) and compute the SLOPE (factor in Artisan) and the INTERCEPT (offset in Artisan). A simple linear transformation.

**Comment by djbetterly** [comment]
Regarding the slope / intercept, you lost me on that one.  I vaguely recall
that from my high school days, but you lost me.  Any chance of getting a
quick primer?

**Reply by MAKOMO** [reply]
Just Google for it. There are plenty of primers online.

**Comment by djbetterly** [comment]
One last thing, and then I'll quit bugging you.  So I input the values
you gave me, and I charted the values of gas pressure (kpa) and
voltage between 0-100 in Artisan. The max value in Artisan for full
gas pressure is 75 and the minimum (maintaining flame) is 30.  Is
there anyway to offset those values so its 0-100?  Or do they need to
exist as 30-75 on the slider?

I also worked out the intercept and slope..thank you.

**Reply by MAKOMO** [reply]
Well, you obviously did not work that out fully.

If you reached from min to max with the OUT1002 with the above settings (offset/intercept=0, factor/slope=0.05) the module did output 1.5V @ slider=30 (30*0.05+0=1.5) and 3.75V @ slider75 (75*0.05+0=3.75).

So now you would like to have Artisan to instruct the OUT1002 to output 1.5V at slider=0 and 3.75V at slider=100, right? So you do again the computation of the intercept and slope (as you find instructions online for Excel and other calculation programs) using as source values 0 and 100 and target values 1.5 and 3.75. This will result in an offset/intercept of 1.5 and a factor/slope of 0.0225.

You can verify that a slider value of 0 now outputs exactly those 1.5V (0 * 0.0225+1.5=1.5) and a slider value of 100 those 3.75V (100 * 0.0225+1.5=3.75).

**Comment by djbetterly** [comment]
ahh..I see what I did wrong now...I was using all of the values from 0-100
& 0-5v.  Thank you!

**Comment by djbetterly** [comment]
Sorry, one more thing, is the offset won't let me do 1.5, only 1 or
2...is that correct?

**Comment by MAKOMO** [comment]
https://shop.artisan.plus/en/consulting/video-conference

**Comment by djbetterly** [comment]
Wow...got it.

**Comment by djbetterly** [comment]
It seems logical that I could take those numbers and multiply by 10, to
give me a whole number, but that doesn't work.  Just trying to find
answers..heres the spreadsheet I created.
